Source/SwiftLintFramework/Rules/ValidDocsRule.swift

```diff
@@ -38,7 +38,11 @@ func superfluousOrMissingThrowsDocumentation(declaration: String, comment: Strin
 }
 
 func delcarationReturns(declaration: String, kind: SwiftDeclarationKind) -> Bool {
-    return declaration.containsString("->") || SwiftDeclarationKind.variableKinds().contains(kind)
+    let outsideBraces = NSMutableString(string: declaration)
+    regex("(\\s*->\\s*).*[^(]\\)").replaceMatchesInString(outsideBraces, options: [],
+        range: NSRange(location: 0, length: outsideBraces.length), withTemplate: "")
+
+    return outsideBraces.containsString("->") || SwiftDeclarationKind.variableKinds().contains(kind)
```

Comment:
Could you return early if `SwiftDeclarationKind.variableKinds().contains(kind)`? Much cheaper than always running the regex replace.